File: Giffer/Models/GIFConfiguration.swift
import CoreGraphics

struct GIFConfiguration {
    var resolutionScale: CGFloat = 1.0
    var fps: Int = 12
    var trimStart: Double = 0.0
    var trimEnd: Double = 1.0
    var cropRect: CGRect? = nil
    var playbackMode: PlaybackMode = .forward

    var trimRange: ClosedRange<Double> {
        trimStart...trimEnd
    }
}

File: android/app/src/main/java/com/leaptools/giffer/model/GifConfiguration.kt
package com.leaptools.giffer.model

import androidx.compose.ui.geometry.Rect

/**
 * All values are normalized (0..1) where they describe a fraction of the source, mirroring
 * the iOS GIFConfiguration. [cropRect] is in normalized image coordinates with the origin at
 * the top-left; null means no crop.
 */
data class GifConfiguration(
    val resolutionScale: Float = 1.0f,
    val fps: Int = 12,
    val trimStart: Double = 0.0,
    val trimEnd: Double = 1.0,
    val cropRect: Rect? = null,
    val playbackMode: PlaybackMode = PlaybackMode.FORWARD,
)
